sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample09_ObservableEventBatchLiveTests.cs

```diff
@@ -0,0 +1,220 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs.Consumer;
+using Azure.Messaging.EventHubs.Producer;
+using NUnit.Framework;
+
+namespace Azure.Messaging.EventHubs.Tests.Snippets
+{
+    /// <summary>
+    ///   The suite of tests defining the snippets used in the Event Hubs
+    ///   Sample09_EventHubsClients sample.
+    /// </summary>
+    ///
+    [TestFixture]
+    [Category(TestCategory.Live)]
+    [Category(TestCategory.DisallowVisualStudioLiveUnitTesting)]
+    [SuppressMessage("Style", "IDE0059:Unnecessary assignment of a value", Justification = "Example assignments needed for snippet output content.")]
+    public class Sample09_ObservableEventBatchLiveTests
+    {
+        /// <summary>
+        ///   Performs basic smoke test validation of the contained snippet.
+        /// </summary>
+        ///
+        [Test]
+        public async Task Sample09_AccessingEventData()
+        {
+            await using var scope = await EventHubScope.CreateAsync(1);
+
+            #region Snippet:Sample09_AccessingEventData
+#if SNIPPET
+            var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+            var eventHubName = "<< NAME OF THE EVENT HUB >>";
+#else
+            var connectionString = EventHubsTestEnvironment.Instance.EventHubsConnectionString;
+            var eventHubName = scope.EventHubName;
+#endif
+
+            var producer = new EventHubProducerClient(connectionString, eventHubName);
+
+            try
+            {
+                using var eventBatch = await producer.CreateBatchAsync();
+                ObservableEventDataBatch newbatch = new ObservableEventDataBatch(eventBatch);
+
+                for (var index = 0; index < 5; ++index)
```

Comment:
I wonder if we should have a small one-liner for a comment before the "we're adding this to the batch"  on L54 and the "we're iterating events in the batch just to show you how to access them" on L65.   My thought is that it may not be apparent that L65-68 are only for illustration and people may wonder why they would want to add things and then immediately loop through them.

Thoughts?

Reply:
Yes I agree, especially in this case since I didn't include the contains check it's not clear.